src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -524,5 +557,52 @@ private EditingOrderViewModel GetEditingOrderViewModel()
                 })
             };
         }
+
+        public async Task<OrderFulfillmentState> CancelOrder(CancelOrderParam param)
+        {
+            if(param == null) throw new ArgumentNullException(nameof(param));
+            if (param.CultureInfo == null) throw new ArgumentException(GetMessageOfNull(nameof(param.CultureInfo)));
+            if (string.IsNullOrWhiteSpace(param.OrderNumber)) throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.OrderNumber)));
+            if (string.IsNullOrWhiteSpace(param.Scope)) throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.Scope)));
+            if (param.CustomerId == Guid.Empty) throw new ArgumentException(GetMessageOfEmpty(nameof(param.CustomerId)));
+
+            var order = await OrderRepository.GetOrderAsync(new GetCustomerOrderParam
+            {
+                Scope = param.Scope,
+                OrderNumber = param.OrderNumber,
+                CustomerId = param.CustomerId
+            }).ConfigureAwait(false);
+            var orderId = Guid.Parse(order.Id);
```

Comment:
here you can have null reference exception
need to move after where you do check if(order ==null)